rfc-156-auto-merge-internal-prs.md

```diff
@@ -110,93 +110,81 @@ Therefore, we propose writing an additional `validate_not_a_bank_holiday` step i
+- `govuk_publishing_components` (18 versions - not including 3 major increases - between October 2022 and January 2023, across 32 apps)
+- `govuk_app_config` (6 versions - not including 1 major release - between October 2022 and January 2023)
+- `gds-api-adapters` (3 major versions between October 2022 and January 2023. 0 minor/patch ones. However, between July and August 2022 there were 4 qualifying patch increases we could have automated.)
+- `rubocop-govuk` (2 versions - 0 major ones - between October 2022 and January 2023, across a large number of repositories)
 
-The PR will be expanded from the initial spike, to include both patch and minor version bumps, of the following internal dependencies (but only where those dependencies are present in the repo):
+For `govuk_publishing_components` alone, we've had 576 PRs between October 2022 and January 2023, all of which could have been auto mergeable. If we estimate a minute of due diligence per PR, that amounts to almost 10 hours of wasted dev time.
 
-- `govuk_publishing_components` (18 versions - not including 3 major increases - since October 2022, across 32 apps)
-- `govuk_app_config` (6 versions - not including 1 major release - since October 2022)
-- `gds-api-adapters` (3 major versions since October 2022. 0 minor/patch ones. However, between July and August 2022 there were 4 qualifying patch increases we could have automated.)
-- `rubocop-govuk` (2 versions - 0 major ones - since October 2022, across a large number of repositories)
+A possible implementation of the configuration file is below, but this is subject to change:
 
-For `govuk_publishing_components` alone, we've had 576 PRs since October 2022, all of which could have been auto mergeable. If we estimate a minute of due diligence per PR, that amounts to almost 10 hours of wasted dev time.
+```yaml
+# /.govuk_auto_merge_pr.yml
+auto_approve_and_merge:
+  version: 1
+  merge_dependencies:
+    - govuk_publishing_components
+    - rubocop-govuk
+  merge_versions:
+    - patch
+    - minor
+```
 
-### Review in 3 months
+### Raise initial config PRs
+
+Following a successful trial, Platform Reliability will raise PRs for all applicable GOV.UK repos. Each PR will configure Dependabot to run during office hours, and create the auto-merge configuration file.
+
+Whilst Platform Reliability would create the initial PRs, it would be up to the owning teams to decide whether or not to merge, and to what extent they'd want to reconfigure the repo and version scopes. We will stipulate that teams must not include external dependencies on the auto-merge list, and must not enable auto-merging without also restricting the Dependabot schedule times.
```

Comment:
It would be good to monitor configuration drift to indicate problems/improvements with this process.